### ISSUE TITLE
unsubscriptable-object should observe py-version for `list[int]`

### ISSUE THREAD

**Issue by uzzell** [pylint-dev/pylint#8954]
### Bug description

Pylint reports an `unsubscriptable-object` message for a type hint that specifies the type of a list (e.g., `list[int]`).

```
"""Unsubscriptable object example"""
x: list[int] = []

```

### Command used

```shell
pylint my_file.py
```

### Pylint output

```shell
************* Module my_file
my_file.py:1:3: E1136: Value 'list' is unsubscriptable (unsubscriptable-object)
```

### Expected behavior

No error message

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.8.10 (default, May 26 2023, 14:05:08)
[GCC 9.4.0]
```

**Comment by jacobtylerwalls** [comment]
Thanks for the report. No warning is emitted if you use a python 3.9 interpreter (and this code is not valid on python 3.8 anyway). However, we have a flag for declaring "minimum supported python version for the code being linted", and it's not taken into account for this message. So accepting as an enhancement to quiet `unsubscriptable-object` for this case when `--py-version=3.9`.

However, we'll be dropping support for python 3.8 in approximately a year when it goes EOL, so there is a low likelihood of someone contributing a patch in time to provide any value, unfortunately.

**Comment by jacobtylerwalls** [comment]
Closing since pylint 3.3 will require a python 3.9+ interpreter.
